client-app/ui-kit/composables/useQuantityValidationSchema.ts

```diff
@@ -3,32 +3,65 @@ import { useI18n } from "vue-i18n";
 import { number } from "yup";
 import type { NumberSchema } from "yup";
 
-export function useQuantityValidationSchema(minQuantity?: number, maxQuantity?: number) {
+export function useQuantityValidationSchema(payload: {
+  minQuantity?: number;
+  maxQuantity?: number;
+  availableQuantity?: number;
+}) {
   const { t } = useI18n();
 
+  const maxLimit = 999999;
+
+  function setMinMaxForSchema(schema: NumberSchema, minQuantity?: number, maxQuantity?: number): NumberSchema {
+    if (minQuantity && maxQuantity) {
+      return schema.test(
+        "minMaxValue",
+        t("shared.cart.add_to_cart.errors.min_max", [minQuantity, maxQuantity]),
+        (value) => !!value && value >= minQuantity! && value <= maxQuantity!,
+      );
+    }
+
+    if (minQuantity) {
+      return schema.min(minQuantity, t("shared.cart.add_to_cart.errors.min", [minQuantity]));
+    }
+
+    if (maxQuantity) {
+      return schema.max(maxQuantity, t("shared.cart.add_to_cart.errors.max", [maxQuantity]));
+    }
+
+    return schema.max(maxLimit, t("shared.cart.add_to_cart.errors.max", [maxLimit]));
+  }
+
   const quantitySchema = computed<NumberSchema>(() =>
     number()
       .typeError(t("shared.cart.add_to_cart.errors.enter_correct_number_message"))
-      .integer()
       .positive()
+      .integer()
       .withMutation((schema) => {
-        if (minQuantity && maxQuantity) {
+        if (payload.availableQuantity && payload.minQuantity) {
           return schema.test(
             "minMaxValue",
-            t("shared.cart.add_to_cart.errors.min_max", [minQuantity, maxQuantity]),
-            (value) => !!value && value >= minQuantity && value <= maxQuantity,
+            t("shared.cart.add_to_cart.errors.min_max", [payload.minQuantity, payload.availableQuantity]),
+            (value) => !!value && value >= payload.minQuantity! && value <= payload.availableQuantity!,
```

Comment:
also no need "!" in the end

Reply:
It leads warnings `Object is possibly undefined`